flybywiresim/a32nx
# fix: API connection logic

Fixes #2746

## Summary of Changes
Check if already connected with the same FLT NBR.
Clean disconnect on FLT NBR change

## References
Not applicable

## Additional context


Discord username (if different from GitHub): nistei#1362

## Testing instructions
1. Find a FLT NBR that is not in use (Check: `https://api.flybywiresim.com/txcxn/_find?flight=YOUR_FLT_NBR`)
2. Create a SimBrief OFP with an airline and flight number
![image](https://user-images.githubusercontent.com/1652722/103163074-41205980-47f9-11eb-9784-c430ebeb6997.png)
3. Set the same combination in the ATC options
![image](https://user-images.githubusercontent.com/1652722/103163086-74fb7f00-47f9-11eb-8c75-dbfcdff3d812.png)
4. Spawn in on the runway
5. MCDU->INIT->INIT REQUEST
6. Should connect to API without issues (Check: `https://api.flybywiresim.com/txcxn/_find?flight=YOUR_FLT_NBR`). Also, the destination/origin should appear in the API result (dest/orig can take up to 30s to update).
7. Change the FLT NBR in the MCDU to another free one
8. Should connect to API without issues
9. Old FLT NBR should be inactive on the API
10. Change the FLT NBR in the MCDU to one that is in use (Pick one from the live map `https://staging.flybywiresim.com/`)
11. Should not connect and show "FLT NBR IN USE". Old FLT NBR should still be active on the API

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link
